src/chrome/content/rules/JianDan.xml

```diff
@@ -1,29 +1,35 @@
 <!--
-
 	mismatch:
-
-		- cdn
+		- g
 		- s
+		
+		But equal to ^ :
+		- cdn
 		- www
 		
 	MCB:
-	  - ^
-
+		- ^
 -->
 
 <ruleset name="JianDan (partial)">
 
 	<target host="jiandan.net" />
-	  <exclusion pattern="^http://jiandan\.net/(?![\w/-]+\.(gif|png))" />
-	  <test url="http://jiandan.net/" />
-	  <test url="http://jiandan.net/new" />
-	  <test url="http://jiandan.net/page/2" />
-	  <test url="http://jandan.net/2016/06/03/fucking-kids.html" />
-	  <test url="http://jandan.net/ofk.gif" />
-	  <test url="http://jandan.net/share/2016/06/qr-79613.png" />
+	<target host="www.jiandan.net" />
+		<test url="http://jandan.net/ofk.gif" />
+		<test url="http://jandan.net/share/2016/06/qr-79613.png" />
+		<test url="http://www.jandan.net/ofk.gif" />
+		<test url="http://www.jandan.net/share/2016/06/qr-79613.png" />
+	<exclusion pattern="^http://(www\.)?jiandan\.net/(?![\w/-]+\.(gif|png))" />
+	<rule from="^http://(www\.)?jiandan\.net/(.+)" to="https://jiandan.net/$1" />
+
+	<target host="cdn.jiandan.net" />
+		<test url="http://cdn.jandan.net/static/img/favicon.ico" />
+		<test url="http://cdn.jandan.net/static/js/common.js" />
+		<test url="http://cdn.jandan.net/wp-content/themes/egg/style.css" />
+	<rule from="^http://cdn\.jiandan\.net/(.+)" to="https://jiandan.net/$1" />
 
 	<target host="tankr.net" />
```

Comment:
Also not that there is different content served on https than on http (not sure if it matters).